server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/CICSVisitor.java

```diff
@@ -103,53 +165,11 @@ public List<Node> visitCicsExecBlock(CICSParser.CicsExecBlockContext ctx) {
             errors.add(error);
         }
 
-        boolean isReturn =
-                (ctx.allCicsRule() != null
-                        && ctx.allCicsRule().size() > 0
-                        && ctx.allCicsRule().get(0).cics_return() != null);
-        boolean isHandle =
-                (ctx.allCicsRule() != null
-                        && ctx.allCicsRule().size() > 0
-                        && ctx.allCicsRule().get(0).cics_handle() != null);
-
-        if (isReturn) {
-            return addTreeNode(ctx, ExecCicsReturnNode::new);
-        } else if (isHandle) {
-            boolean isProgram =
-                    Optional.ofNullable(ctx.allCicsRule().get(0).cics_handle())
-                            .map(CICSParser.Cics_handleContext::cics_handle_abend)
-                            .map(CICSParser.Cics_handle_abendContext::PROGRAM)
-                            .filter(s -> s.size() > 0)
-                            .isPresent();
-
-            boolean isLabel =
-                    Optional.ofNullable(ctx.allCicsRule().get(0).cics_handle())
-                            .map(CICSParser.Cics_handleContext::cics_handle_abend)
-                            .map(CICSParser.Cics_handle_abendContext::LABEL)
-                            .filter(s -> s.size() > 0)
-                            .isPresent();
-
-            boolean isReset =
-                    Optional.ofNullable(ctx.allCicsRule().get(0).cics_handle())
-                            .map(CICSParser.Cics_handleContext::cics_handle_abend)
-                            .map(CICSParser.Cics_handle_abendContext::RESET)
-                            .filter(s -> s.size() > 0)
-                            .isPresent();
-
-            ExecCicsHandleNode.HandleAbendType type;
-            if (isProgram) {
-                type = ExecCicsHandleNode.HandleAbendType.PROGRAM;
-            } else if (isLabel) {
-                type = ExecCicsHandleNode.HandleAbendType.LABEL;
-            } else if (isReset) {
-                type = ExecCicsHandleNode.HandleAbendType.RESET;
-            } else {
-                type = ExecCicsHandleNode.HandleAbendType.CANCEL;
-            }
-
-            return addTreeNode(ctx, (location) -> new ExecCicsHandleNode(location, type));
+        List<Node> nodes = visitChildren(ctx);
+        if (nodes.isEmpty()) {
+            return addTreeNode(ctx, ExecCicsNode::new);
         }
-        return addTreeNode(ctx, ExecCicsNode::new);
+        return nodes;
```

Comment:
I don't understand the reason for this change.

Reply:
Okay, what I must do?

Reply:
I think the original code was doing the right thing - it created the execcicsnode and attached children nodes:
```suggestion
        return addTreeNode(ctx, ExecCicsNode::new);
```

Reply:
But there are no children, because all EXEC CICS statement are on the same level, and I just moved this code to different methods to simplify the class, but now to obtain proper locality we need to point to the EXEC CICS statement, that's why we have to call parent 3 times